packages/frontend/src/components/ChartConfigPanel/Grid/index.tsx

```diff
@@ -1,55 +1,130 @@
+import { Button, FormGroup, InputGroup } from '@blueprintjs/core';
 import { EchartsGrid } from '@lightdash/common';
-import React, { FC } from 'react';
-import { useForm } from 'react-hook-form';
+import startCase from 'lodash/startCase';
+import { FC, useState } from 'react';
 import { useVisualizationContext } from '../../LightdashVisualization/VisualizationProvider';
-import Form from '../../ReactHookForm/Form';
-import Input from '../../ReactHookForm/Input';
-import { SectionRow, SectionTitle } from './Grid.styles';
+import { SectionRow } from './Grid.styles';
 
-export const defaultGrid: EchartsGrid = {
+export const defaultGrid = {
     containLabel: true,
     left: '5%', // small padding
     right: '5%', // small padding
-    top: '70', // pixels from top (makes room for legend)
-    bottom: '30', // pixels from bottom (makes room for x-axis)
+    top: '70px', // pixels from top (makes room for legend)
+    bottom: '30px', // pixels from bottom (makes room for x-axis)
+};
+
+const positions = ['top', 'bottom', 'left', 'right'] as const;
+
+enum Units {
+    Pixels = 'px',
+    Percentage = '%',
+}
+
+const getNextUnit = (current?: Units): Units | undefined => {
+    if (!current) return;
+
+    const units = Object.values(Units);
+    const currentIndex = units.indexOf(current);
+    return units.concat(units[0])[currentIndex + 1];
+};
+
+const getValueAndUnit = (valueWithUnit?: string): [string?, Units?] => {
+    if (!valueWithUnit || valueWithUnit === '') return [];
+
+    const unit =
+        Object.values(Units).find((u) => valueWithUnit.endsWith(u)) ||
+        Units.Pixels;
+
+    const value = valueWithUnit.replace(unit, '');
+    return [value, unit];
 };
 
 const GridPanel: FC = () => {
     const {
         cartesianConfig: { dirtyEchartsConfig, setGrid },
     } = useVisualizationContext();
-    const methods = useForm<EchartsGrid>({
-        mode: 'onBlur',
-        defaultValues: { ...defaultGrid, ...dirtyEchartsConfig?.grid },
-    });
+
+    const config = useMemo<EchartsGrid>(
```

Comment:
useMemo needs to be imported  

![Screenshot 2022-11-15 at 09 16 38](https://user-images.githubusercontent.com/9117144/201879709-398a09b9-bd9c-4262-9645-70f9a4a1bd12.png)